layers/+tools/docker/packages.el

```diff
@@ -52,8 +52,15 @@
     :config
     (spacemacs/declare-prefix-for-mode 'dockerfile-mode "mc" "compile")
     (spacemacs/set-leader-keys-for-major-mode 'dockerfile-mode
+      "b" 'dockerfile-build-buffer
+      "B" 'dockerfile-build-buffer-no-cache-buffer
       "cb" 'dockerfile-build-buffer
-      "cB" 'dockerfile-build-no-cache-buffer)))
+      "cB" 'dockerfile-build-no-cache-buffer)
+    (with-eval-after-load 'docker
+      (spacemacs/set-leader-keys-for-major-mode 'dockerfile-mode
+        "d" 'docker
+        "i" 'docker-images
+        "p" 'docker-containers))))
```

Comment:
By wrapping them in `with-eval-after-load`, these keybindings will not be accessible if `docker` is not already loaded beforehand. There may be use-cases where one spins up a new Emacs instance just to edit a Dockerfile without loading the `docker` package. I suggest that you move these under `:init` since they are autoloaded by the `docker` package anyways.

Reply:
A couple questions:
Independent of `docker` package being not yet loaded at some point, wouldn't the keybindings be accessible anyway as soon as the package is gets to load? (Thus effectively only delaying the key bindings until the last moment, but never skipping them)
What if the user excludes the `docker` package via `dotspacemacs-excluded-packages`? (This is the main corner case that made me wrap everything in `with-eval-after-load`).

Reply:
Finally learnt how to do it properly. Fixed on #15707.